### PR TITLE
Execute any release job on any PR

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2730

### CODE REVIEW

**Review by bbguimaraes — COMMENTED** [review]
The duplication here is unfortunate, can't we create a single, uniform list that is processed by the old code?

Just as one example, should we support `AggregateCount` for these new jobs?

File: cmd/payload-testing-prow-plugin/server.go [hunk 1]
```diff
@@ -32,7 +33,10 @@ type githubClient interface {
 	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
 }
 
-var ocpPayloadTestsPattern = regexp.MustCompile(`(?mi)^/payload\s+(?P<ocp>4\.\d+)\s+(?P<release>\w+)\s+(?P<jobs>\w+)\s*$`)
+var (
+	ocpPayloadTestsPattern    = regexp.MustCompile(`(?mi)^/payload\s+(?P<ocp>4\.\d+)\s+(?P<release>\w+)\s+(?P<jobs>\w+)\s*$`)
+	ocpPayloadJobTestsPattern = regexp.MustCompile(`(?mi)^/payload-job(?P<jobs>(\s+(\w|\.)+(-(\w|\.)+)*)+)\s*$`)
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go:38]: > You can also place the \s+ outside the capture and remove the TrimSpace call below.
I cannot make this right.

File: cmd/payload-testing-prow-plugin/server.go [hunk 2]
```diff
@@ -189,6 +218,38 @@ func (s *server) handle(l *logrus.Entry, ic github.IssueCommentEvent) string {
...
+			})
+		}
+
+		run := builder.build(releaseJobSpecsFromComment)
+		if err := s.kubeClient.Create(s.ctx, run); err != nil {
+			logger.WithError(err).Error("could not create PullRequestPayloadQualificationRun")
+			return formatError(fmt.Errorf("could not create PullRequestPayloadQualificationRun: %w", err))
+		}
+		messages = append(messages, message(jobSetSpecification{}, jobNames))
+		messages = append(messages, fmt.Sprintf("See details on %s/%s/%s\n", prPayloadTestsUIURL, builder.namespace, run.Name))
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go:249]: Let me try again.

[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go]: Done.

File: cmd/payload-testing-prow-plugin/server.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbguimaraes) [reply cmd/payload-testing-prow-plugin/server.go]:
```go
re := regexp.MustCompile(`(?mi)^/payload-job\s+((?:[-\w.]+\s*?)+)\s*$`)
s := "/payload-job    abc    def    ghi-4.10   "
for _, m := range re.FindAllStringSubmatch(s, -1) {
	fmt.Printf("%#v\n", strings.Fields(m[1]))
}
```
```
[]string{"abc", "def", "ghi-4.10"}
```
(note that `Fields` conforms more to the `\s`s in the pattern)

[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/payload-testing-prow-plugin/server.go]: Nice. `((?:re)+)` and `\s*?` I will remember these patterns.
You made me learn regex again.

[the remark this answers is not in the document]
Comment (bbguimaraes) [reply cmd/payload-testing-prow-plugin/server.go]: It's a dark art, but sometimes useful =)

**Comment by bbguimaraes** [comment]
That was a very nice improvement.

/lgtm

File: cmd/payload-testing-prow-plugin/server.go [hunk 3]
```diff
@@ -32,7 +32,10 @@ type githubClient interface {
 	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
 }
 
-var ocpPayloadTestsPattern = regexp.MustCompile(`(?mi)^/payload\s+(?P<ocp>4\.\d+)\s+(?P<release>\w+)\s+(?P<jobs>\w+)\s*$`)
+var (
+	ocpPayloadTestsPattern    = regexp.MustCompile(`(?mi)^/payload\s+(?P<ocp>4\.\d+)\s+(?P<release>\w+)\s+(?P<jobs>\w+)\s*$`)
```
Comment (droslean) [inline cmd/payload-testing-prow-plugin/server.go:36]: Can we have unittests for these patterns?
